### PR TITLE
Add Feather M0 Board

### PR DESCRIPTION
**Pull request by torntrousers**
Add a new target for the Adafruit Feather M0 - https://learn.adafruit.com/adafruit-feather-m0-basic-proto/pinouts

Tested with Blinky1, UART0 with console output, UART1 with GPS input, and I2C with an SHT31

### CODE REVIEW

**Comment by aykevl** [comment]
Can you rebase onto the dev branch? There are currently some conflicts.

**Comment by torntrousers** [comment]
Isn't it already? ```hms-tinygo git:(featherM0) ✗ git rebase dev
Current branch featherM0 is up to date.```

**Comment by aykevl** [comment]
Not according to GitHub, did you pull the dev branch?

**Comment by aykevl** [comment]
Also, there seem to be many Feather M0 boards and only 3 of them contain a LoRa module. What about renaming it `feather-m0-lora` or something like that?
@deadprogram any better idea?

**Comment by torntrousers** [comment]
I think I've now properly merged the latest dev branch.
I thought all the Feather M0 would share these common definitions, though I'm not 100% sure.

**Comment by aykevl** [comment]
Maybe the problem is that you're merging instead of rebasing?
Try this:
```
git checkout dev
git pull origin dev
git checkout featherM0
git rebase dev
git push -f
```

**Comment by deadprogram** [comment]
There is nothing really specific to LoRa in this PR. 

On the Feather M0 variation boards there are additional chips, but the processor and pins are the same.

We can best handle that case, by providing a separate TinyGo driver for whatever additional SPI device is included onboard. This will also allow people who have acquired a separate breakout board with the same addition device.

What do you think?

**Comment by deadprogram** [comment]
Note also this branch can be probably best be merged by squashing/merging.

**Comment by deadprogram** [comment]
Thanks for the PR @torntrousers now merging.

Please note that I am migrating the SAM family of boards to use the new `volatile` package/API. Take a look at https://github.com/tinygo-org/tinygo/compare/dev...feature/sam-refactor-volatile to see how it works.

All of the real work is already done, so perhaps you can test that branch once I merge this PR into it.

Thanks again for another great contribution!

**Comment by deadprogram** [comment]
OK that branch now has the Feather M0. Also the Trinket M0. Perhaps you can verify those boards still work as you expect. I've successfully tested the SAMD21 boards that I myself have.

File: src/machine/board_feather-m0.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// +build sam,atsamd21,itsybitsy_m0
```
Comment (aykevl) [inline src/machine/board_feather-m0.go]: This line caused the build to fail. It must be:

```go
// +build sam,atsamd21,feather_m0
```

Comment (torntrousers) [reply src/machine/board_feather-m0.go]: Sorry. I did fix that but my dev env is a bit of a mess and the code I run isn't in the same folder as the git checkout so I need to copy changes back and forth and sometimes lose things. I need to sort that out.

Comment (deadprogram) [reply src/machine/board_feather-m0.go]: I already fixed in `dev` branch, so no further work on this PR required.

**Pull request merged**